Crashes when importing xray_repository_config

if one types

`terraform import xray_repository_config.xray-repo-config collection-container-prod-local`

one gets
```
xray_repository_config.xray-repo-config: Importing from ID "collection-container-prod-local"...
xray_repository_config.xray-repo-config: Import prepared!
  Prepared xray_repository_config for import
xray_repository_config.xray-repo-config: Refreshing state... [id=collection-container-prod-local]
╷
│ Error: Request cancelled
│
│ The plugin.(*GRPCProvider).ReadResource request was cancelled.
╵


Stack trace from the terraform-provider-xray_v1.15.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x1052d3290]

goroutine 41 [running]:
github.com/jfrog/terraform-provider-xray/pkg/xray.resourceXrayRepositoryConfig.func8(0x140001111d0, {0x14000452f00, 0x6}, {0x14000307aa0, 0x6})
	github.com/jfrog/terraform-provider-xray/pkg/xray/resource_xray_repository_config.go:376 +0x170
github.com/jfrog/terraform-provider-xray/pkg/xray.resourceXrayRepositoryConfig.func12({0x1400051a3c0?, 0x14000338160?}, {{0x140001bf9c0?, 0x10530030b?}, 0x140001111d0?, 0x0?}, 0xd?, {0x14000452f00, 0x6}, {0x14000307aa0, ...})
	github.com/jfrog/terraform-provider-xray/pkg/xray/resource_xray_repository_config.go:429 +0x108
github.com/jfrog/terraform-provider-xray/pkg/xray.resourceXrayRepositoryConfig.func14({0x1055b2268, 0x140005485a0}, 0x140001e0900, {0x105526d00?, 0x1400047e870})
	github.com/jfrog/terraform-provider-xray/pkg/xray/resource_xray_repository_config.go:482 +0x568
github.com/jfrog/terraform-provider-shared/util/sdk.applyTelemetry.func1({0x1055b2268?, 0x140005485a0}, 0x0?, {0x105526d00?, 0x1400047e870})
	github.com/jfrog/terraform-provider-shared@v1.18.0/util/sdk/sdk.go:234 +0x200
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x14000498c40, {0x1055b22a0, 0x140002e8fc0}, 0xd?, {0x105526d00, 0x1400047e870})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/resource.go:750 +0xec
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0x14000498c40, {0x1055b22a0, 0x140002e8fc0}, 0x140001971e0, {0x105526d00, 0x1400047e870})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/resource.go:1044 +0x494
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0x14000317968, {0x1055b22a0?, 0x140002e8ea0?}, 0x140004b2b80)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/grpc_provider.go:616 +0x40c
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0x140004b4000, {0x1055b22a0?, 0x140002e8360?}, 0x140005482a0)
	github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov5/tf5server/server.go:751 +0x3ec
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x10557b900?, 0x140004b4000}, {0x1055b22a0, 0x140002e8360}, 0x140001dc070, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:386 +0x174
google.golang.org/grpc.(*Server).processUnaryRPC(0x1400046e000, {0x1055b4ee0, 0x140002a89c0}, 0x140004f2000, 0x1400047f3b0, 0x105a4df50, 0x0)
	google.golang.org/grpc@v1.56.1/server.go:1337 +0xc7c
google.golang.org/grpc.(*Server).handleStream(0x1400046e000, {0x1055b4ee0, 0x140002a89c0}, 0x140004f2000, 0x0)
	google.golang.org/grpc@v1.56.1/server.go:1714 +0x840
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.56.1/server.go:959 +0x88
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.56.1/server.go:957 +0x170

Error: The terraform-provider-xray_v1.15.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

```


The terraform for it is trivial:

```
terraform {
  required_providers {
    xray = {
      source = "jfrog/xray"
      version = "1.15.0"
    }
  }
}

provider "xray" {
   url = "https://XXXXX.jfrog.io"
}

resource "xray_repository_config" "xray-repo-config" {
  repo_name = "anyname"
}
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,4 +1,4 @@
-## 1.16.0 (September 7, 2023)
+## 1.16.0 (September 7, 2023). Tested on Artifactory 7.63.14 and Xray 3.81.8
 
 IMPROVEMENTS:
 * resource/xray_repository_config: added validation to ensure either `config` or `path_config` attribute is defined. 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,11 @@
+## 1.16.0 (September 7, 2023)
+
+IMPROVEMENTS:
+* resource/xray_repository_config: added validation to ensure either `config` or `path_config` attribute is defined. 
+
+PR: [#135](https://github.com/jfrog/terraform-provider-xray/pull/135)
+Issue: [#134](https://github.com/jfrog/terraform-provider-xray/issues/134)
+
 ## 1.15.0 (August 29, 2023). Tested on Artifactory 7.63.14 and Xray 3.80.9
 
 BUG FIXES:

--- a/pkg/xray/resource_xray_repository_config.go
+++ b/pkg/xray/resource_xray_repository_config.go
@@ -78,11 +78,11 @@ func resourceXrayRepositoryConfig() *schema.Resource {
 			ValidateDiagFunc: validator.StringIsNotEmpty,
 		},
 		"config": {
-			Type:          schema.TypeSet,
-			Optional:      true,
-			MaxItems:      1,
-			Description:   "Single repository configuration. Only one of 'config' or 'paths_config' can be set.",
-			ConflictsWith: []string{"paths_config"},
+			Type:         schema.TypeSet,
+			Optional:     true,
+			MaxItems:     1,
+			Description:  "Single repository configuration. Only one of 'config' or 'paths_config' can be set.",
+			AtLeastOneOf: []string{"paths_config"},
 			Elem: &schema.Resource{
 				Schema: map[string]*schema.Schema{
 					"vuln_contextual_analysis": {
@@ -141,42 +141,43 @@ func resourceXrayRepositoryConfig() *schema.Resource {
 			},
 		},
 		"paths_config": {
-			Type:        schema.TypeSet,
-			Optional:    true,
-			MaxItems:    1,
-			Description: `Enables you to set a more granular retention period. It enables you to scan future artifacts within the specific path, and set a retention period for the historical data of artifacts after they are scanned`,
+			Type:         schema.TypeSet,
+			Optional:     true,
+			MaxItems:     1,
+			Description:  "Enables you to set a more granular retention period. It enables you to scan future artifacts within the specific path, and set a retention period for the historical data of artifacts after they are scanned",
+			AtLeastOneOf: []string{"config"},
 			Elem: &schema.Resource{
 				Schema: map[string]*schema.Schema{
 					"pattern": {
 						Type:        schema.TypeList,
 						Required:    true,
 						MinItems:    1,
-						Description: `Pattern, applied to the repositories.`,
+						Description: "Pattern, applied to the repositories.",
 						Elem: &schema.Resource{
 							Schema: map[string]*schema.Schema{
 								"include": {
 									Type:             schema.TypeString,
 									Required:         true,
-									Description:      `Include pattern.`,
+									Description:      "Include pattern.",
 									ValidateDiagFunc: validator.StringIsNotEmpty,
 								},
 								"exclude": {
 									Type:             schema.TypeString,
 									Optional:         true,
-									Description:      `Exclude pattern.`,
+									Description:      "Exclude pattern.",
 									ValidateDiagFunc: validator.StringIsNotEmpty,
 								},
 								"index_new_artifacts": {
 									Type:        schema.TypeBool,
 									Optional:    true,
 									Default:     true,
-									Description: `If checked, Xray will scan newly added artifacts in the path. Note that existing artifacts will not be scanned. If the folder contains existing artifacts that have been scanned, and you do not want to index new artifacts in that folder, you can choose not to index that folder.`,
+									Description: "If checked, Xray will scan newly added artifacts in the path. Note that existing artifacts will not be scanned. If the folder contains existing artifacts that have been scanned, and you do not want to index new artifacts in that folder, you can choose not to index that folder.",
 								},
 								"retention_in_days": {
 									Type:             schema.TypeInt,
 									Optional:         true,
 									Default:          90,
-									Description:      `The artifact will be retained for the number of days you set here, after the artifact is scanned. This will apply to all artifacts in the repository.`,
+									Description:      "The artifact will be retained for the number of days you set here, after the artifact is scanned. This will apply to all artifacts in the repository.",
 									ValidateDiagFunc: validator.IntAtLeast(0),
 								},
 							},
@@ -185,7 +186,7 @@ func resourceXrayRepositoryConfig() *schema.Resource {
 					"all_other_artifacts": {
 						Type:        schema.TypeSet,
 						Required:    true,
-						Description: `If you select by pattern, you must define a retention period for all other artifacts in the repository in the All Other Artifacts setting.`,
+						Description: "If you select by pattern, you must define a retention period for all other artifacts in the repository in the All Other Artifacts setting.",
 						MinItems:    1,
 						MaxItems:    1,
 						Elem: &schema.Resource{
@@ -194,13 +195,13 @@ func resourceXrayRepositoryConfig() *schema.Resource {
 									Type:        schema.TypeBool,
 									Optional:    true,
 									Default:     true,
-									Description: `If checked, Xray will scan newly added artifacts in the path. Note that existing artifacts will not be scanned. If the folder contains existing artifacts that have been scanned, and you do not want to index new artifacts in that folder, you can choose not to index that folder.`,
+									Description: "If checked, Xray will scan newly added artifacts in the path. Note that existing artifacts will not be scanned. If the folder contains existing artifacts that have been scanned, and you do not want to index new artifacts in that folder, you can choose not to index that folder.",
 								},
 								"retention_in_days": {
 									Type:             schema.TypeInt,
 									Optional:         true,
 									Default:          90,
-									Description:      `The artifact will be retained for the number of days you set here, after the artifact is scanned. This will apply to all artifacts in the repository.`,
+									Description:      "The artifact will be retained for the number of days you set here, after the artifact is scanned. This will apply to all artifacts in the repository.",
 									ValidateDiagFunc: validator.IntAtLeast(0),
 								},
 							},
@@ -372,7 +373,7 @@ func resourceXrayRepositoryConfig() *schema.Resource {
 			"retention_in_days": repoConfig.RetentionInDays,
 		}
 
-		if slices.Contains(vulnContextualAnalysisPackageTypes(xrayVersion), packageType) {
+		if repoConfig.VulnContextualAnalysis != nil && slices.Contains(vulnContextualAnalysisPackageTypes(xrayVersion), packageType) {
 			m["vuln_contextual_analysis"] = *repoConfig.VulnContextualAnalysis
 		}
 
@@ -528,5 +529,4 @@ func resourceXrayRepositoryConfig() *schema.Resource {
 		Schema:      repositoryConfigSchema,
 		Description: "Provides an Xray repository config resource. See [Xray Indexing Resources](https://www.jfrog.com/confluence/display/JFROG/Indexing+Xray+Resources#IndexingXrayResources-SetaRetentionPeriod) and [REST API](https://www.jfrog.com/confluence/display/JFROG/Xray+REST+API#XrayRESTAPI-UpdateRepositoriesConfigurations) for more details.",
 	}
-
 }

--- a/pkg/xray/resource_xray_repository_config_test.go
+++ b/pkg/xray/resource_xray_repository_config_test.go
@@ -11,23 +11,20 @@ import (
 	"github.com/jfrog/terraform-provider-shared/util/sdk"
 )
 
-func TestAccRepositoryConfigRepoConfigNegative(t *testing.T) {
+func TestAccRepositoryConfigRepoNoConfig(t *testing.T) {
 	_, fqrn, resourceName := testutil.MkNames("xray-repo-config-", "xray_repository_config")
 	var testData = map[string]string{
-		"resource_name":                resourceName,
-		"repo_name":                    "repo-config-test-repo",
-		"retention_in_days":            "90",
-		"pattern0_include":             "core/**",
-		"pattern0_exclude":             "core/external/**",
-		"pattern0_index_new_artifacts": "true",
-		"pattern0_retention_in_days":   "45",
-		"pattern1_include":             "core/**",
-		"pattern1_exclude":             "core/external/**",
-		"pattern1_index_new_artifacts": "true",
-		"pattern1_retention_in_days":   "45",
-		"other_index_new_artifacts":    "true",
-		"other_retention_in_days":      "60",
+		"resource_name": resourceName,
+		"repo_name":     "repo-config-test-repo",
 	}
+
+	config := sdk.ExecuteTemplate(
+		fqrn,
+		`resource "xray_repository_config" "{{ .resource_name }}" {
+			repo_name = "{{ .repo_name }}"
+		}`,
+		testData,
+	)
 
 	resource.Test(t, resource.TestCase{
 		PreCheck:          func() { testAccPreCheck(t) },
@@ -35,8 +32,8 @@ func TestAccRepositoryConfigRepoConfigNegative(t *testing.T) {
 
 		Steps: []resource.TestStep{
 			{
-				Config:      sdk.ExecuteTemplate(fqrn, TestDataRepoConfigErrorTemplate, testData),
-				ExpectError: regexp.MustCompile("Conflicting configuration arguments"),
+				Config:      config,
+				ExpectError: regexp.MustCompile("Missing required argument"),
 			},
 		},
 	})
@@ -115,7 +112,7 @@ func testAccRepositoryConfigRepoConfigCreate_VulnContextualAnalysis(t *testing.T
 	}
 }
 
-func TestAccRepositoryConfigRepoConfigCreate_Exposure(t *testing.T) {
+func TestAccRepositoryConfigRepoConfigCreate_exposure(t *testing.T) {
 	testCase := []struct {
 		packageType  string
 		template     string
@@ -179,6 +176,37 @@ func TestAccRepositoryConfigRepoConfigCreate_Exposure(t *testing.T) {
 	}
 }
 
+func TestAccRepositoryConfigRepoConfigCreate_no_exposure(t *testing.T) {
+	packageTypes := []string{"alpine", "bower", "composer", "conan", "conda", "debian", "gems", "generic", "go", "gradle", "ivy", "nuget", "rpm", "sbt"}
+	template := `
+	resource "xray_repository_config" "{{ .resource_name }}" {
+		repo_name = "{{ .repo_name }}"
+
+		config {
+			retention_in_days = {{ .retention_in_days }}
+		}
+	}`
+	validVersion := "3.75.10"
+	version, err := sdk.GetXrayVersion(GetTestResty(t))
+	if err != nil {
+		t.Fail()
+		return
+	}
+
+	checkFunc := func(fqrn string, testData map[string]string) resource.TestCheckFunc {
+		return resource.ComposeTestCheckFunc(
+			resource.TestCheckResourceAttr(fqrn, "repo_name", testData["repo_name"]),
+			resource.TestCheckResourceAttr(fqrn, "config.0.retention_in_days", testData["retention_in_days"]),
+			resource.TestCheckNoResourceAttr(fqrn, "config.0.vuln_contextual_analysis"),
+			resource.TestCheckResourceAttr(fqrn, "config.0.exposures.#", "0"),
+		)
+	}
+
+	for _, packageType := range packageTypes {
+		t.Run(packageType, testAccRepositoryConfigRepoConfigCreate(t, packageType, template, validVersion, version, checkFunc))
+	}
+}
+
 func testAccRepositoryConfigRepoConfigCreate(t *testing.T, packageType, template, validVersion, xrayVersion string, checkFunc func(fqrn string, testData map[string]string) resource.TestCheckFunc) func(t *testing.T) {
 	return func(t *testing.T) {
 		_, fqrn, resourceName := testutil.MkNames("xray-repo-config-", "xray_repository_config")
@@ -211,7 +239,6 @@ func testAccRepositoryConfigRepoConfigCreate(t *testing.T, packageType, template
 				return errorResp, err
 			}),
 			ProviderFactories: testAccProviders(),
-
 			Steps: []resource.TestStep{
 				{
 					Config: sdk.ExecuteTemplate(fqrn, template, testData),
@@ -377,36 +404,6 @@ func verifyRepositoryConfig(fqrn string, testData map[string]string) resource.Te
 	)
 }
 
-const TestDataRepoConfigErrorTemplate = `
-resource "xray_repository_config" "{{ .resource_name }}" {
-  repo_name = "{{ .repo_name }}"
-
-  config {
-    retention_in_days = {{ .retention_in_days }} 
-  }
-
-  paths_config {
-    pattern {
-      include             = "{{ .pattern0_include }}"
-      exclude             = "{{ .pattern0_exclude }}"
-      index_new_artifacts = {{ .pattern0_index_new_artifacts }}
-      retention_in_days   = {{ .pattern0_retention_in_days }}
-    }
-
-    pattern {
-      include             = "{{ .pattern1_include }}"
-      exclude             = "{{ .pattern1_exclude }}"
-      index_new_artifacts = {{ .pattern1_index_new_artifacts }}
-      retention_in_days   = {{ .pattern1_retention_in_days }}
-    }
-
-    all_other_artifacts {
-      index_new_artifacts = {{ .other_index_new_artifacts }}
-      retention_in_days   = {{ .other_retention_in_days }}
-    }
-  }
-}`
-
 const TestDataRepoConfigDockerTemplate = `
 resource "xray_repository_config" "{{ .resource_name }}" {
   repo_name = "{{ .repo_name }}"
@@ -446,7 +443,7 @@ resource "xray_repository_config" "{{ .resource_name }}" {
   repo_name = "{{ .repo_name }}"
 
   config {
-    retention_in_days        = {{ .retention_in_days }}
+    retention_in_days = {{ .retention_in_days }}
 
 	exposures {
       scanners_category {

